quarry/web/worker.py

```diff
@@ -148,6 +149,22 @@ def run_query(query_run_id):
             celery_log.info("Stopped run for qrun:%s", qrun.id)
         else:
             write_error(qrun, e.args[1])
+    except Exception as e:
+        qrun.status = QueryRun.STATUS_FAILED
+        qrun.extra_info = json.dumps({"error": e.args[1]})
+        conn.session.add(qrun)
+        conn.session.commit()
+        tb_str = "".join(
+            traceback.format_exception(
+                etype=type(e), value=e, tb=e.__traceback__
```

Comment:
This works in python 3.7 (current version in prod), but fails on python 3.10 with `TypeError: format_exception() got an unexpected keyword argument 'etype'`. Just `traceback.format_exception(e)` gives the same output in python 3.10.

Please add a comment noting the above as we'll need to update it while upgrading python.

Reply:
I've changed this to check current python version using `sys.version_info` and format accordingly, to avoid any inconvenience when we upgrade python.